pipelines/imu/readers.py

```diff
@@ -199,13 +198,17 @@ def extract_data(filename: str, packet: Dict[str, Dict]) -> xr.Dataset:
     return xr.Dataset.from_dict(dictionary)
 
 
-class ImuFileHandler(AbstractFileHandler):
-    def write(self, ds: xr.Dataset, filename: str, config: Config, **kwargs):
-        raise NotImplementedError(
-            "Error: this file format should not be used to write to."
-        )
+class IMUDataReader(DataReader):
+    class Parameters(BaseModel, extra=Extra.forbid):
+        """If your CustomDataReader should take any additional arguments from the
+        retriever configuration file, then those should be specified here.
+
+        e.g.,:
+        custom_parameter: float = 5.0
+
+        """
 
-    def read(self, filename: str, **kwargs) -> xr.Dataset:
+    def read(self, filename: str, **kwargs) -> Union[xr.Dataset, Dict[str, xr.Dataset]]:
```

Comment:
`filename` should be renamed to `input_key` to be consistent with the tsdat read signature for `DataReader` objects. This applies for anywhere else DataReaders are used in this PR too